pulsar-client-cpp/lib/ConsumerImpl.h

```diff
@@ -166,6 +170,18 @@ class ConsumerImpl : public ConsumerImplBase,
 
     MessageCryptoPtr msgCrypto_;
 
+    Optional<MessageId> lastMessageInBroker_;
```

Comment:
Rather than an optional, just initialized to MessageId(). You can avoid the logic below in lastMessageIdAvailable

Reply:
Thanks, this is try to align with other startMessageId_/lastDequedMessage_  in this file.